core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -68,6 +68,10 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
         final MediaType contentType = request.contentType();
         if (contentType != null && (contentType.is(MediaType.JSON) ||
                                     contentType.subtype().endsWith("+json"))) {
+            if (expectedResultType == String.class) {
```

Comment:
In [JSON specifications](https://tools.ietf.org/html/rfc7159.html#page-4), a JSON string value, such as `"something"`, can also be a JSON text. So this might make a wrong result that a user doesn't expect.
So how about trying to convert the content to a JSON object and passing it to the next `RequestConverterFunction` if it failed? 
In this way, you might inject the content as a byte array as well. (as described in the original issue #1584) Also, it may handle XML document, too. 
Please let me know if my comment is unclear. 😃